examples/pingmessage/pingmessage.ino

```diff
@@ -148,8 +152,35 @@ void loop()
             break;
         }
 
-        toggleLed();
-        printf("> id: %d\t Length: %d\t parsed: %d\t errors: %d", parser.rxMsg.message_id(), parser.rxMsg.payload_length(),
-              parser.parsed, parser.errors);
+        Serial.print("> id: ");
+        Serial.print(parser.rxMsg.message_id());
+        Serial.print("\t Length: ");
+        Serial.print(parser.rxMsg.payload_length());
+        Serial.print("\t parsed: ");
+        Serial.print(parser.parsed);
+        Serial.print("\t errors: ");
+        Serial.println(parser.errors);
     }
+
+    // Toggle the LED to show that the program is running
+    digitalWrite(ledPin, !digitalRead(ledPin));
+}
+
+bool waitMessage(enum Ping1DNamespace::msg_ping1D_id id, uint16_t timeout_ms = 400)
+{
+    uint32_t tstart = millis();
```

Comment:
you can create a single timeout variable here.
```cpp
 uint32_t timeout_time = millis() + timeout_ms;
 while (millis() < timeout_time) {
```

Reply:
this amounts to exactly the same thing. I don't think it's easier to read.

Reply:
done

Reply:
For each while loop it needs to run one more operation to do the + operation

Reply:
I wonder if the compiler is smarter than me.